src/app/hooks/useConversationCache.ts

```diff
@@ -0,0 +1,56 @@
+import { useCache, useEventManager } from 'react-components';
+import createCache from 'proton-shared/lib/helpers/cache';
+import createLRU from 'proton-shared/lib/helpers/lru';
+
+import { Event } from '../models/eventManager';
+import { EVENT_ACTIONS } from 'proton-shared/lib/constants';
+import { omit } from 'proton-shared/lib/helpers/object';
+import { ConversationResult } from './useConversation';
+import { Cache } from '../models/utils';
+
+const CACHE_KEY = 'ConversationsCache';
+
+export type ConversationCache = Cache<string, ConversationResult>;
+
+const conversationListener = (cache: ConversationCache) => ({ Conversations }: Event) => {
+    if (!Array.isArray(Conversations)) {
+        return;
+    }
+
+    for (const { ID, Action, Conversation } of Conversations) {
+        // Ignore updates for non-fetched messages.
+        if (!cache.has(ID)) {
+            continue;
+        }
+        if (Action === EVENT_ACTIONS.DELETE) {
+            cache.delete(ID);
+        }
+        if (Action === EVENT_ACTIONS.UPDATE_DRAFT) {
+            console.warn('Event type UPDATE_DRAFT on Conversation not supported', Conversations);
+        }
+        if (Action === EVENT_ACTIONS.UPDATE_FLAGS) {
+            const currentValue = cache.get(ID);
+
+            cache.set(ID, {
+                Conversation: {
+                    ...currentValue,
+                    ...omit(Conversation, ['LabelIDsRemoved', 'LabelIDsAdded'])
+                },
+                Messages: currentValue.Messages
+            });
+        }
+    }
+};
+
+export const useConversationCache = (): ConversationCache => {
+    const globalCache = useCache();
+    const { subscribe } = useEventManager();
+
+    if (!globalCache.has(CACHE_KEY)) {
+        const cache = createCache(createLRU({ max: 50 } as any));
+        subscribe(conversationListener(cache));
```

Comment:
⚠️  Right now it's subscribing each time the component renders! Side effects like subscribe has to happen in a useEffect and the callback to unsubscribe has to be called in the cleanup callback

Reply:
Well, it could be more elegant but it's not subscribing at each render. The `if (!globalCache.has(CACHE_KEY))` should allow the cache creation to be ran only once during the whole execution.

I thought about unsubscribing. But right now, I'm not sure we have to stop listening to events in the app.

Well, if we consider merging with `proton-mail-settings` then yes, message / conversation logic should perhaps be unloaded at one point.

Reply:
I remember why I didn't use a `useEffect` to subscribe to events. It would subscribe for each component using the cache.

If we want to be able to unsubscribe, I fear it will require to come back to a provider.

Reply:
Yes my bad, you are right, it would only subscribe each time the cache is created.

In the future we might want to change the global cache for an already logged in user. Another case would be logging out (however the event manager already cleans up zombie listeners), or as you say when we merge with proton-mail-settings. 

When using `subscribe` it should respect that it should call unsubscribe when it's no longer needed. It's not that much code to add? And it makes more safe and future-proof.